### DISCUSSION TITLE
How to implement variable width horizontal sorting list properly?

### DISCUSSION THREAD

**Discussion by samesfahani-tuplehealth** [clauderic/dnd-kit#713] [Q&A] [answered]
https://user-images.githubusercontent.com/66535998/165170541-7417c8cb-79ae-48bc-a945-47b1d54fe24e.mov

Apologies if this is a very basic ask, but I am at a loss as to what's wrong here. As you can see, the widths are not being calculated properly and the transforms are all off. I am really trying to replicate the `Sortable Horizontal Variable Widths` story a la https://5fc05e08a4a65d0021ae0bf2-ipdskkittb.chromatic.com/?path=/story/presets-sortable-horizontal--variable-widths&globals=backgrounds.grid:false

Here is my repro:
https://codesandbox.io/s/vigorous-brown-k63m6x?file=/src/DraggableHeader.tsx

**Comment by samesfahani-tuplehealth** [comment, marked as answer]
I can see that I was incorrectly using the index as the id. Just logical errors in the onDragStart/End handlers, in case that helps anyone.

**Comment by remitoffoli** [comment]
Hi @samesfahani-tuplehealth, thank you for sharing your Typescript code! 🙂

I've been struggling for 3 hours trying to get a simple vertical sortable component. Yours was the first example I saw using objects, not strings, as items. 🙌 And now I understand we MUST use an "id" property as item identifier. 🤦‍♂️ I now have to find a way to convert my data's "uids". 🤷‍♂️
